### ISSUE TITLE
smtp.SMTP would be faster using LineOnlyReceiver

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11132]
|<img alt="nickm's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| nickm reported|
|-|-|
|Trac ID|trac#1602|
|Type|enhancement|
|Created|2006-03-24 08:09:43Z|

I tried to write a load generator for $PRODUCT, and found that smtp.SMTP was too
slow, but that I could improve the performance by over 3x (which was enough to
stress the $PRODUCT) by switching smtp.SMTP's dataReceived() with one close to
LineReceiver's.  Is there anything keeping this from going into the mainline?

Attachments:

* [ticket1602.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/61f/61f348dbfc99215070417b9d49da662cfd0671ef/1ea6658ac661fea1e381587e5e10ba8689c92ce1.diff) (828 bytes) - added by PenguinOfDoom on 2006-03-24 08:28:49Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1602 1602
type__enhancement enhancement
reporter__nickm nickm
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1143187783000000 1143187783000000
changetime__1143338018000000 1143338018000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|PenguinOfDoom commented|
|-|-|

Tests pass with the attached patch. Does Divmod use smtp.SMTP.rawDataReceived for something?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16457]) Change t.mail.smtp.SMTP to use LineOnlyReceiver instead of LineReceiver

Author: exarkun
Reviewer: itamarst
Fixes [#11132](https://github.com/twisted/twisted/issues/11132)

SMTP is a purely line-oriented protocol.  The existing code never places SMTP
into raw-mode, and there is no good reason for anyone to ever do this.  Delete
rawDataReceived and change the base class to the faster LineOnlyReceiver.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
